### ISSUE TITLE
The model keeps converging!

### ISSUE THREAD

**Issue by wanmeiziyoulan** [Tianxiaomo/pytorch-YOLOv4#202]
Hi, thanks for your share! 
In fact, I find the loss is very large when I traning. I tried to change the learning rate, but it didn't work. The dataset I used is coco.
Do you have any advice?

 ```
2020-07-30 21:22:50,064 train.py[line:616] INFO: Using device cuda
  2 2020-07-30 21:24:26,585 train.py[line:330] INFO: Starting training:
  3         Epochs:          300
  4         Batch size:      64
  5         Subdivisions:    16
  6         Learning rate:   0.001
  7         Training size:   4028
  8         Validation size: 5
  9         Checkpoints:     True
 10         Device:          cuda
 11         Images size:     608
 12         Optimizer:       adam
 13         Dataset classes: 2
 14         Train label path:data/sum/train.txt
 15         Pretrained:
 16 
 17 2020-07-30 21:41:38,060 train.py[line:414] DEBUG: Train step_320: loss : 66611.1796875,loss xy : 87.65125274658203,loss wh : 15.129814147949219,loss obj : 66462.8125，loss cls : 45.587646484375,los
    s l2 : 24424.3828125,lr : 1.30321e-10
 18 2020-07-30 22:04:48,702 train.py[line:414] DEBUG: Train step_640: loss : 66522.328125,loss xy : 34.19620895385742,loss wh : 6.0429792404174805,loss obj : 66463.3671875，loss cls : 18.71764373779297
    ,loss l2 : 24384.033203125,lr : 2.3134409999999997e-09
 19 2020-07-30 22:17:53,828 train.py[line:414] DEBUG: Train step_960: loss : 66581.578125,loss xy : 46.234989166259766,loss wh : 13.603473663330078,loss obj : 66498.3515625，loss cls : 23.3903503417968
    75,loss l2 : 24418.90234375,lr : 1.2117360999999998e-08
 20 2020-07-30 22:20:11,475 train.py[line:449] INFO: Created checkpoint directory
 21 2020-07-30 22:20:12,056 train.py[line:454] INFO: Checkpoint 1 saved !
 22 2020-07-30 22:23:35,664 train.py[line:414] DEBUG: Train step_1280: loss : 66615.4609375,loss xy : 43.62785339355469,loss wh : 7.414181232452393,loss obj : 66543.09375，loss cls : 21.331085205078125
    ,loss l2 : 24437.75390625,lr : 3.8950081e-08
 23 2020-07-30 22:26:17,570 train.py[line:414] DEBUG: Train step_1600: loss : 66606.8125,loss xy : 60.55643844604492,loss wh : 13.483831405639648,loss obj : 66503.1171875，loss cls : 29.65764617919922,
    loss l2 : 24424.365234375,lr : 9.605960100000002e-08
 24 2020-07-30 22:28:57,618 train.py[line:414] DEBUG: Train step_1920: loss : 66420.46875,loss xy : 33.17624282836914,loss wh : 4.751721382141113,loss obj : 66364.65625，loss cls : 17.880712509155273,l
    oss l2 : 24332.041015625,lr : 2.0053392099999997e-07
 25 2020-07-30 22:29:48,761 train.py[line:449] INFO: Created checkpoint directory
 26 2020-07-30 22:29:49,283 train.py[line:454] INFO: Checkpoint 2 saved !
 27 2020-07-30 22:33:25,488 train.py[line:414] DEBUG: Train step_2240: loss : 66553.2265625,loss xy : 115.07807922363281,loss wh : 19.84312629699707,loss obj : 66356.4375，loss cls : 61.86677169799805,
    loss l2 : 24376.373046875,lr : 3.7330104100000016e-07
 28 2020-07-30 22:36:11,632 train.py[line:414] DEBUG: Train step_2560: loss : 66648.2265625,loss xy : 74.88021087646484,loss wh : 10.111303329467773,loss obj : 66525.96875，loss cls : 37.26540374755859
    4,loss l2 : 24430.205078125,lr : 6.391289610000001e-07
 29 2020-07-30 22:38:47,309 train.py[line:414] DEBUG: Train step_2880: loss : 66563.1796875,loss xy : 53.3154182434082,loss wh : 5.115115642547607,loss obj : 66477.8515625，loss cls : 26.89851570129394
    5,loss l2 : 24388.40625,lr : 1.026625681e-06
 30 2020-07-30 22:39:57,280 train.py[line:449] INFO: Created checkpoint directory
 31 2020-07-30 22:39:57,795 train.py[line:454] INFO: Checkpoint 3 saved !
 32 2020-07-30 22:41:57,019 train.py[line:414] DEBUG: Train step_3200: loss : 66581.6953125,loss xy : 80.40172576904297,loss wh : 73.53779602050781,loss obj : 66386.0，loss cls : 41.75857925415039,loss
     l2 : 24486.849609375,lr : 1.5682392010000005e-06
 33 2020-07-30 22:44:39,062 train.py[line:414] DEBUG: Train step_3520: loss : 66625.4765625,loss xy : 79.5181884765625,loss wh : 29.602218627929688,loss obj : 66475.8046875，loss cls : 40.5532608032226
    56,loss l2 : 24450.044921875,lr : 2.300257521e-06
 34 2020-07-30 22:47:32,278 train.py[line:414] DEBUG: Train step_3840: loss : 66594.703125,loss xy : 59.08452606201172,loss wh : 10.338085174560547,loss obj : 66494.7109375，loss cls : 30.5711135864257
    8,loss l2 : 24424.130859375,lr : 3.2628086409999994e-06
 35 2020-07-30 22:49:27,799 train.py[line:449] INFO: Created checkpoint directory
 36 2020-07-30 22:49:28,336 train.py[line:454] INFO: Checkpoint 4 saved !
 37 2020-07-30 22:51:34,987 train.py[line:414] DEBUG: Train step_4160: loss : 66604.265625,loss xy : 166.27996826171875,loss wh : 23.156702041625977,loss obj : 66331.640625，loss cls : 83.1876296997070
    3,loss l2 : 24394.54296875,lr : 4.499860561e-06
 38 2020-07-30 22:54:21,672 train.py[line:414] DEBUG: Train step_4480: loss : 66595.3984375,loss xy : 75.24266815185547,loss wh : 13.074333190917969,loss obj : 66467.6484375，loss cls : 39.430591583251
    95,loss l2 : 24413.724609375,lr : 6.0592212810000025e-06
 39 2020-07-30 22:57:06,301 train.py[line:414] DEBUG: Train step_4800: loss : 66639.46875,loss xy : 64.50112915039062,loss wh : 25.16973114013672,loss obj : 66516.1015625，loss cls : 33.69256591796875,
    loss l2 : 24466.841796875,lr : 7.992538800999999e-06
 40 2020-07-30 22:59:08,817 train.py[line:449] INFO: Created checkpoint directory
 41 2020-07-30 22:59:09,441 train.py[line:454] INFO: Checkpoint 5 saved !
```

**Comment by PPGod95** [comment]
I used the coins dataset, and the loss decrease after 30-40th epoch 


`2020-07-30 02:30:30,203 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:30:32,687 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 39 saved !
2020-07-30 02:31:06,968 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_200: loss : 62389.9921875,loss xy : 135.1375732421875,loss wh : 34.70897674560547,loss obj : 62118.73828125，loss cls : 101.40568542480469,loss l2 : 22255.716796875,lr : 1.6000000000000004e-06
2020-07-30 02:31:09,292 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:31:11,702 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 40 saved !
2020-07-30 02:31:50,245 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:31:52,687 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 41 saved !
2020-07-30 02:32:10,366 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_208: loss : 62144.44921875,loss xy : 110.2764892578125,loss wh : 9.931215286254883,loss obj : 61940.23828125，loss cls : 84.00474548339844,loss l2 : 22127.232421875,lr : 1.8717736959999997e-06
2020-07-30 02:32:27,320 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:32:29,687 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 42 saved !
2020-07-30 02:33:15,033 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:33:17,689 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 43 saved !
2020-07-30 02:33:22,407 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_216: loss : 62185.10546875,loss xy : 144.89349365234375,loss wh : 30.40617561340332,loss obj : 61900.0390625，loss cls : 109.76368713378906,loss l2 : 22162.072265625,lr : 2.176782336e-06
2020-07-30 02:33:50,362 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:33:52,689 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 44 saved !
2020-07-30 02:34:21,775 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_224: loss : 62055.4140625,loss xy : 165.91871643066406,loss wh : 55.79731750488281,loss obj : 61710.171875，loss cls : 123.52571868896484,loss l2 : 22130.62109375,lr : 2.5176309760000004e-06
2020-07-30 02:34:28,647 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:34:30,687 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 45 saved !
2020-07-30 02:34:59,952 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:35:02,693 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 46 saved !
2020-07-30 02:35:15,441 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_232: loss : 61852.6484375,loss xy : 122.04171752929688,loss wh : 21.332176208496094,loss obj : 61615.19921875，loss cls : 94.07567596435547,loss l2 : 21993.904296875,lr : 2.8970229760000003e-06
2020-07-30 02:35:38,804 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:35:40,686 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 47 saved !
2020-07-30 02:36:14,426 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_240: loss : 61628.8515625,loss xy : 132.00311279296875,loss wh : 23.618587493896484,loss obj : 61372.3125，loss cls : 100.91880798339844,loss l2 : 21870.046875,lr : 3.3177599999999996e-06
2020-07-30 02:36:16,824 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:36:18,685 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 48 saved !
2020-07-30 02:37:02,008 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:37:05,661 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 49 saved !
2020-07-30 02:37:29,241 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_248: loss : 61808.49609375,loss xy : 156.82940673828125,loss wh : 29.23468017578125,loss obj : 61505.40234375，loss cls : 117.03033447265625,loss l2 : 21973.1328125,lr : 3.782742016e-06
2020-07-30 02:37:48,198 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:37:50,688 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 50 saved !
2020-07-30 02:38:28,501 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:38:31,688 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 51 saved !
2020-07-30 02:38:38,001 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_256: loss : 61462.55078125,loss xy : 140.18516540527344,loss wh : 25.414697647094727,loss obj : 61187.81640625，loss cls : 109.13321685791016,loss l2 : 21798.380859375,lr : 4.294967296000001e-06
2020-07-30 02:39:07,687 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:39:09,693 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 52 saved !
2020-07-30 02:39:36,677 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_264: loss : 61093.05859375,loss xy : 178.0669403076172,loss wh : 43.75157928466797,loss obj : 60740.03515625，loss cls : 131.20285034179688,loss l2 : 21629.841796875,lr : 4.857532416000001e-06
2020-07-30 02:39:45,392 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:39:47,690 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 53 saved !
2020-07-30 02:40:25,515 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:40:27,687 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 54 saved !
2020-07-30 02:40:40,238 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_272: loss : 60799.26171875,loss xy : 143.86444091796875,loss wh : 23.42880630493164,loss obj : 60522.40234375，loss cls : 109.5672836303711,loss l2 : 21489.0625,lr : 5.473632256000001e-06
2020-07-30 02:41:02,145 <ipython-input-1-4c51901897dc>[line:446] INFO: Created checkpoint directory
2020-07-30 02:41:04,685 <ipython-input-1-4c51901897dc>[line:451] INFO: Checkpoint 55 saved !
2020-07-30 02:41:42,736 <ipython-input-1-4c51901897dc>[line:411] DEBUG: Train step_280: loss : 60674.4765625,loss xy : 143.0797119140625,loss wh : 44.220096588134766,loss obj : 60381.60546875，loss cls : 105.57003784179688,loss l2 : 21446.791015625,lr : 6.146560000000002e-06`

**Comment by wanmeiziyoulan** [comment]
@PPGod95 Thanks for your reply. Your results are also kept at a large loss. It's a normal?

**Comment by PPGod95** [comment]
Yes, I trained 300 epochs and the loss dropped to 1000. You can also see [https://github.com/Tianxiaomo/pytorch-YOLOv4/blob/master/Use_yolov4_to_train_your_own_data.md](url).

**Comment by wanmeiziyoulan** [comment]
OK, thank you! I'll  train more epochs to see how it works.

**Comment by ZhangYang0726** [comment]
batchsize 给小一点就行了
